src/mds/Server.cc

```diff
@@ -1118,6 +1118,15 @@ void Server::find_idle_sessions()
 	     << "), not marking any client stale" << dendl;
     return;
   }
+  // don't kick client if ops or osds are laggy
+  if(mds->objecter->with_osdmap([](const OSDMap &map)
+  { return map.are_osds_laggy(); })) {
```

Comment:
ditto

Reply:
do we need to check for laggy osds here? we already have done it in `evict_cap_revoke_non_responders()` and as per xiubo we might not need it here. check this conversation https://github.com/ceph/ceph/pull/49971/commits/150abaf856815f99afda0959872fc9350472669e#r1111455604

Reply:
I think both is correct. You may not have a laggy client but it's forced to delay cap releases by laggy OSDs. You may also have the reverse situation: no cap releases expected by the MDS but the client is unresponsive.

Reply:
right, it' good to have both here, thanks!

Reply:
@batrick - PR https://github.com/ceph/ceph/pull/53578 introduces back some changes that evicts clients in find_idle_session() rather than deferring them if the OSDs are laggy. This is discussed (with Dhairya) in this comment: https://github.com/ceph/ceph/pull/53578#issuecomment-1748885300

I feel find_idle_sessions() should not have checks for laggy OSDs and it suffices to do that in evict_cap_revoke_non_responders(), but I see you proposed it to be included in find_idle_sessions().